### ISSUE TITLE
Magento 2.1.0: An quote item of a grouped product does not have the correct tax class applied after it was added to the quote

### ISSUE THREAD

**Issue by heldchen** [magento/magento2#5844]
when adding a **child product of a grouped produc**t into the empty cart, the resulting Magento\Quote\Model\Quote\Item has wrong tax and price informations. after the quote is edited/saved (f.e. by adding another item) both tax and prices are correct, even for the first product. 

simple and configurable products are not affected from this bug.
### Preconditions
1. magento 2.1.0 (official package) with some grouped products that have non-zero tax percent classes (in our case: 2.5% sales tax on all items for all users regardless of location)
2. tax settings (cart & orders):
3. display prices: including tax
4. display subtotal: including tax
5. display shipping amount: including tax
6. include tax in order total: no
7. display full tax summary: no
8. display zero tax subtotal: no
### Steps to reproduce

add some code to output the current quote items values, f.e. to the cart phtml:

```
$checkoutSession = $this->_objectManager->get('\Magento\Checkout\Model\Session');
$quote = $checkoutSession->getQuote();

foreach ($quote->getAllItems() as $item) {
    echo '<br />sku: '.$item->getSku();
    echo '<br />tax%: '.$item->getTaxPercent();
    echo '<br />price: '.$item->getPrice();
    echo '<br />price+tax: '.$item->getPriceInclTax();
}
```
1. empty cart for an anonymous user
2. add a child product of a grouped product
3. observe cart debug output on /checkout/cart/
4. add another product to the cart (type does not matter)
5. observe cart debug output on /checkout/cart/
6. remove second item
7. observe cart debug output on /checkout/cart/
8. add a simple product to the cart
9. observe cart debug output on /checkout/cart/ 
### Expected result

**the expected output after first item:**

```
sku: grouped1
tax%: 2.5000
price: 38.8300
price+tax: 39.8000
```

**the expected output after second item added:**

```
sku: grouped1
tax%: 2.5000
price: 38.8300
price+tax: 39.8000
sku: grouped2
tax%: 2.5000
price: 9.7100
price+tax: 9.9500
```

**the expected output after second item removed:**

```
sku: grouped1
tax%: 2.5000
price: 38.8300
price+tax: 39.8000
```

**the expected output after simple item added:**

```
sku: grouped1
tax%: 2.5000
price: 38.8300
price+tax: 39.8000
sku: simple1
tax%: 2.5000
price: 3.7100
price+tax: 3.8000
```
### Actual result

**the output after first item:**

```
sku: grouped1
tax%: 0.0000
price: 39.8000
price+tax: 39.8000
```

→ price & tax for second grouped item is not correct

**the output after second item:**

```
sku: grouped1
tax%: 2.5000
price: 38.8300
price+tax: 39.8000
sku: grouped2
tax%: 0.0000
price: 9.9500
price+tax: 9.9500
```

→ price & tax for first grouped item is correct
→ price & tax for second grouped item is not correct

**the output after second item removed:**

```
sku: grouped1
tax%: 2.5000
price: 38.8300
price+tax: 39.8000
```

→ price & tax for first grouped item is correct

**the expected output after simple item added:**

```
sku: grouped1
tax%: 2.5000
price: 38.8300
price+tax: 39.8000
sku: simple1
tax%: 2.5000
price: 3.7100
price+tax: 3.8000
```

→ price & tax for first grouped item is correct
→ price & tax for simple item is correct

**Comment by heldchen** [comment]
some additional information on this:
- the database record in `quote_item` has wrong values in the fields `tax_percent`, `tax_amount`, `base_tax_amount` (and possibly more) for the **last** item added to the cart if this item has `product_type` of `grouped`
- as such, preliminary probing shows the bug might also affecting at least parts of the later checkout process
- when two different products of the same grouped products are added to the cart in the same step, both quote_item records will have wrong tax & price values
- when a `quote_item` record is updated (i.e. by changing the quantity in the cart) the values are again stored correctly into the database
- child products of configurable products do have correct values in `quote_item`
- in our case, we need the proper tax information for affiliate reasons (i.e. Google Dynamic Remarketing Tag)

**Comment by mfutselaar** [comment]
Has this issue been resolved yet, or is it going to be resolved? I feel it's still around in 2.1.3.

- Tax in checkout is zero when I add a grouped product
- I add a simple product to the cart, tax is added to the totals
- The added tax is the tax for the simple products only, it doesn't include the grouped products

**Comment by heldchen** [comment]
as far as I see it's not resolved.

**Comment by magento-engcom-team** [comment]
@heldchen, thank you for your report.
The issue is already fixed in develop branch, 2.2.1
